File: CAT 1/Rectangle.cpp
#include <iostream>
using namespace std;

class Rectangle {
private:
    // Attributes
    double width;
    double length;

public:
    // Constructor
    Rectangle(double l, double w) : length(l), width(w) {}

    // Setter methods
    void setLength(double l) {
        length = l;
    }

    void setWidth(double w) {
        width = w;
    }

    // Getter methods
    double getLength() const {
        return length;
    }

    double getWidth() const {
        return width;
    }

    // Method to compute area
    double areaOfRect() const {
        return length * width;
    }

    // Method to compute perimeter
    double findPerimeter() const {
        return 2 * (length + width);
    }
};

int main() {
    // Instantiate the Rectangle class
    Rectangle R1(20, 30);
    Rectangle R2(30, 34);

    // Print the area using the instance method without parameters
    cout << "The Area is: " << R2.areaOfRect() << endl;

    // Ask for user input to modify the rectangle's dimensions
    double w, l;
    cout << "Enter the width: ";
    cin >> w;

    cout << "Enter the length: ";
    cin >> l;

    // Set the values using the setters
    R1.setWidth(w);
    R1.setLength(l);

    // Output the updated values
    cout << "Updated width: " << R1.getWidth() << endl;
    cout << "Updated length: " << R1.getLength() << endl;
    cout << "Updated Area: " << R1.areaOfRect() << endl;
    cout << "Updated Perimeter: " << R1.findPerimeter() << endl;

    return 0;
}

File: CAT 1/Rectangle.py
class Rectangle:
    # Constructor
    def __init__(self, length, width):
        self.length = length
        self.width = width

    # Setter methods
    def set_length(self, length):
        self.length = length

    def set_width(self, width):
        self.width = width

    # Getter methods
    def get_length(self):
        return self.length

    def get_width(self):
        return self.width

    # Method to compute the area
    def area_of_rect(self):
        return self.length * self.width

    # Method to compute the perimeter
    def find_perimeter(self):
        return 2 * (self.length + self.width)


# Main block
if __name__ == "__main__":
    # Instantiate the Rectangle class
    R1 = Rectangle(20, 30)
    R2 = Rectangle(30, 34)

    # Print the area using the instance method without parameters
    print("The Area is:", R2.area_of_rect())

    # Ask for user input to modify the rectangle's dimensions
    w = float(input("Enter the width: "))
    l = float(input("Enter the length: "))

    # Set the values using the setters
    R1.set_width(w)
    R1.set_length(l)

    # Output the updated values
    print("Updated width:", R1.get_width())
    print("Updated length:", R1.get_length())
    print("Updated Area:", R1.area_of_rect())
    print("Updated Perimeter:", R1.find_perimeter())